src/main/java/uk/nhs/hee/tis/trainee/credentials/event/ProgrammeMembershipEventListener.java

```diff
@@ -61,7 +65,16 @@ void deleteProgrammeMembership(DeleteEventDto deletedProgrammeMembership) {
   @SqsListener("${application.aws.sqs.update-programme-membership}")
   void updateProgrammeMembership(UpdateEventDto updatedProgrammeMembership) {
     log.info("Received update event for programme membership {}.", updatedProgrammeMembership);
-    // For now, we simply revoke regardless of which fields have updated (pending TIS21-4152)
-    revocationService.revoke(updatedProgrammeMembership.tisId(), CredentialType.TRAINING_PROGRAMME);
+
+    RecordDto recrd = updatedProgrammeMembership.recrd();
+    String programmeName = recrd.getData().get("TPR-programmeName");
+    String startDate = String.valueOf(
+        LocalDate.parse(recrd.getData().get("TPR-PlacementStartDate")));
+    String endDate = String.valueOf(LocalDate.parse(recrd.getData().get("TPR-PlacementEndDate")));
```

Comment:
Switch the variable names to match the update event contents instead of the credential.